src/test/java/com/cloudbees/plugins/credentials/CredentialsInPipelineTest.java

```diff
@@ -0,0 +1,703 @@
+     * Initially tied to JENKINS-70101 research.
+     */
+
+    // For developers: set to `true` so that pipeline console logs show
+    // up in System.out (and/or System.err) of the plugin test run by
+    //   mvn test -Dtest="CredentialsInPipelineTest"
+    private boolean verbosePipelines = false;
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    // Data for build agent setup
+    @Rule
+    public TemporaryFolder tmpAgent = new TemporaryFolder();
+    @Rule
+    public TemporaryFolder tmpWorker = new TemporaryFolder();
+    // Where did we save that file?..
+    private File agentJar = null;
+    // Can this be reused for many test cases?
+    private Slave agent = null;
+    // Unknown/started/not usable
+    private Boolean agentUsable = null;
+
+    // From CertificateCredentialImplTest
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+    private File p12;
+
+    @Before
+    public void setup() {
+        r.jenkins.setCrumbIssuer(null);
+    }
+
+    Boolean isAvailableAgent() {
+        // Can be used to skip optional tests if we know we could not set up an agent
+        if (agentJar == null)
+            return false;
+        if (agent == null)
+            return false;
+        return agentUsable;
+    }
+
+    Boolean setupAgent() throws IOException, InterruptedException, OutOfMemoryError {
+        // Note we anticipate this might fail; it should not block the whole test suite from running
+        // Loosely inspired by
+        // https://docs.cloudbees.com/docs/cloudbees-ci-kb/latest/client-and-managed-masters/create-agent-node-from-groovy
+
+        // Is it known-impossible to start the agent?
+        if (agentUsable != null && agentUsable == false)
+            return agentUsable; // quickly for re-runs
+
+        // Did we download this file for earlier test cases?
+        if (agentJar == null) {
+            try {
+                URL url = new URL(r.jenkins.getRootUrl() + "jnlpJars/agent.jar");
+                agentJar = tmpAgent.newFile("agent.jar");
+                FileOutputStream out = new FileOutputStream(agentJar);
+                out.write(url.openStream().readAllBytes());
+                out.close();
+            } catch (IOException | OutOfMemoryError e) {
+                agentJar = null;
+                agentUsable = false;
+
+                System.out.println("Failed to download agent.jar from test instance: " +
+                        e.toString());
+
+                return agentUsable;
+            }
+        }
+
+        // This CLI spelling and quoting should play well with both Windows
+        // (including spaces in directory names) and Unix/Linux
+        ComputerLauncher launcher = new CommandLauncher(
+                "\"" + System.getProperty("java.home") + File.separator + "bin" +
+                File.separator + "java\" -jar \"" + agentJar.getAbsolutePath().toString() + "\""
+        );
```

Comment:
this creates an memory unconstrained JVM so can gram 1/4 of the system memory. 
`JenkinsRule.createOnlineSlave` handles this and so much more for you (passing through debug args, setting the JVM to headless to avoid dock bouncing on mac etc etc)